server.js

```diff
@@ -79,73 +79,88 @@ http
                                 clientId_1: '',
                             },
                         };
-
-                        transaqConnector.objectAccountsAndDll.servers[HftOrNot] = {
+                        objectAccountsAndDll.servers[HftOrNot] = {
                             host,
                             port,
                         };
-                        return transaqConnector.functionConnect(HftOrNot, data => {
+                    return transaqConnector.functionConnect(HftOrNot, data => {
                             const message = JSON.parse(xml2json.toJson(data));
-
-                            if (!message) {
-                                return;
-                            }
-
-                            if (!message.sec_info_upd && !message.pits && !message.securities) {
-                                console.log(message);
+                             //if message and other info exist
+                            if (!message && !message.sec_info_upd && !message.pits && !message.securities) {
+                              console.log(message);
                             }
-                            if (message.client && message.client.id) {
-                                transaqConnector.objectAccountsAndDll
-                                    .users[HftOrNot].Account = message.client;
-                                transaqConnector.objectAccountsAndDll
-                                    .users[HftOrNot].Account.clientId_1 = message.client.id;
-                            }
-
-                            if (message.messages && message.messages.message && message.messages.message.text === 'Password expired. Please change the password') {
+                            // set value if they exist
+                            transaqConnector.objectAccountsAndDll.users[HftOrNot].Account = !!message.client && !!message.client.id && message.client;
+                            transaqConnector.objectAccountsAndDll.users[HftOrNot].Account.clientId_1 = !!message.client && !!message.client.id && message.client.id;
+                            const incoming_message=message.messages
+                            if (!!incoming_message && !!incoming_message.message && incoming_message.message.text === 'Password expired. Please change the password') {
                                 // TODO: popup about pass expired.
+                          
                                 console.log('pass expired');
                             }
 
                             if (message['server_status']) {
                                 if (message.server_status.connected === 'error' || message.server_status.connected === 'false') {
                                     // TODO: popup about connect error and redirect to login page
+                             
+                                  // redirect to login page
+                                  res.redirect('./login');
```

Comment:
Is this the correct login page?
It's an Ajax request sent by opextrade. The redirect will happen not in UX, but in the API?

Reply:
ok I didn't know that I will correct